martin-observability-utils/src/lib.rs

```diff
@@ -0,0 +1,129 @@
+#![doc = include_str!("../README.md")]
+
+#[derive(Default)]
+pub struct MartinObservability {
```

Comment:
Bit unsure about the API/naming here..

I tried to make it less "magic", but unsure if this is the best way to handle it.